File: spec/javascript/equity_client_spec.rb
# frozen_string_literal: true

require "rails_helper"

RSpec.describe "EquityClient", type: :feature, js: true do

  before do
    create_list(:deal, 3)
  end

  let(:deals) { Deal.all }

  it "works" do
    visit "/"
    puts "*" * 80
    p page.evaluate_script("console.log(2)")
    p page.driver.browser.manage.logs.get("browser")
    puts "*" * 80
    expect(page).to have_content 'Hello'
  end
end

File: spec/javascript/equity_client.spec.js
import EquityClient from 'equity_client'

const api = new EquityClient('http://localhost:3001/api/v1')

describe('ApiClient', () => {
  describe('deals', () => {
    describe('list', () => {
      it('workd', async () => {
        expect.assertions(1)

        const actual = await api.deals.list()
        console.log(actual)
        expect(actual).toBeEqual(1)
      })
    })
  })
})
